README.md

```diff
@@ -218,6 +219,50 @@ Note that First-Party Sets also gives browsers the opportunity to group per-site
 those at `chrome://settings/content/all`) by the “first-party” boundary instead of eTLD+1, which is 
 not always the correct site boundary.
 
+# Clearing Site Data on Set Transitions
+Sites can switch the First-Party Set that they are a member of. We need to pay attention to these transitions so that they don’t link user identities across all the FPSs they’ve historically been in. The concrete goals that we aim to achieve are as follows:
+
+*   Preserve user expectations that their browsing activity is scoped/partitioned to “party” (as defined by First-Party Set).
+*   Prevent a potential privacy attack that uses a throwaway domain to rotate between sets, and therefore sync user data or identifiers across disparate sets. 
+
+In order to achieve this, site data needs to be cleared on certain transitions., The clearing should behave like [Clear-Site-Data](https://www.w3.org/TR/clear-site-data/), for which site data includes cookies, storage, cache, as well as execution contexts (documents, workers, etc.). We don’t differentiate between different types of site data since a site could expect different types of data to co-exist and require some consistency.
+
+Sites switching FPSs can be viewed as “leaving one FPS and joining another”, and the member sites can only add/remove themselves to/from FPSs with the consent from the owners. If a FPS’s owner changes, that set is considered dissolved. Thus we associate the need to clear site data with whether that site’s FPS owner has changed, with a few exceptions.
```

Comment:
> 1. The first clause here is inconsistent with the "Potential modification" below.
Thanks for the suggestion. You are correct that the “Potential modification” is more like a special case for “not clear” when “sites change their owner” with the purpose to avoid adoption pain.

> 2. The second clause, about owner consent, seems irrelevant to when we clear a site's data.
The second clause is to point out that “owner” is important to define a FPS, which hopefully helps introduce the idea “we associate the need to clear site data with whether that site’s FPS owner has changed”. In other words, from the implementation perspective, we know that a site left/joined a set by looking at the owner. Does that make sense? Is it best to not mention the implementation details from the spec perspective?

> 3. The second sentence is also inconsistent with the potential modification.
We can remove the second sentence if that’s causing confusion.

> 
> Maybe just assert that we consider whether to clear site data each time a site's FPS owner changes?
How about:
Sites switching FPSs can be viewed as “leaving one FPS and joining another”, and the member sites can only add/remove themselves to/from FPSs with the consent from the owners. Thus we associate the need to clear site data with whether that site’s FPS owner has changed, with a few exceptions.